src/ast/ast.cpp

```diff
@@ -580,6 +580,11 @@ std::string Probe::name() const
   return n;
 }
 
+std::string Probe::args_typename() const
+{
+  return "struct " + name() + "_args";
```

Comment:
I don't understand the code fully myself yet, but I had a couple of questions about this at first glance:
- Does this name get exposed to users anywhere? e.g. in casting
- Could it conflict with a user-defined type name? e.g. if they have a struct called "input_args" and a function called "input"

Reply:
- If a user tries to cast `args` then the error message will contain this type name:
```
# bpftrace -e 'kfunc:vfs_read { printf("%d\n", ((struct foo)args).count); exit(); }'
stdin:1:33-47: ERROR: Cannot resolve unknown type "struct foo"
kfunc:vfs_read { printf("%d\n", ((struct foo)args).count); exit(); }
                                ~~~~~~~~~~~~~~
stdin:1:33-47: ERROR: Cannot cast from struct type "struct kfunc:vmlinux:vfs_read_args"
kfunc:vfs_read { printf("%d\n", ((struct foo)args).count); exit(); }
```
- `name()` returns the full probe name, so the type would be called `struct uprobe:bin:input_args` which is not a valid C identifier, so it cannot conflict a user type.